source/adapters/level_zero/event.cpp

```diff
@@ -1137,7 +1140,8 @@ ur_result_t urEventReleaseInternal(ur_event_handle_t Event) {
   if (DisableEventsCaching || !Event->OwnNativeHandle) {
     delete Event;
   } else {
-    Event->Context->addEventToContextCache(Event);
```

Comment:
Aren't we leaking the Event now, since it's neither `delete`ed nor added to the cache?

Reply:
I simplified this. I had actually fixed the original memory leak that Winston was addressing here:

https://github.com/oneapi-src/unified-runtime/pull/2476

Please re-review, the patch now is a revert and minor update of the patch from Winston.